.github/workflows/java-windows.yml

```diff
@@ -17,12 +19,38 @@ jobs:
                 with:
                     java-version: ${{ matrix.java_version }}
 
+            -   name: Cache gradle dependencies
+                uses: actions/cache@v1
+                with:
+                    path: ~/.gradle/caches
+                    key: ${{ runner.os }}-gradle-${{matrix.java_version}}-${{ hashFiles('**/*.gradle*') }}
+                    restore-keys: |
+                        ${{ runner.os }}-gradle-
+
+            -   name: Cache maven dependencies
+                uses: actions/cache@v1
+                with:
+                    path: ~/.m2
+                    key: ${{ runner.os }}-maven-${{matrix.java_version}}-${{ hashFiles('**/pom.xml') }}
+                    restore-keys: |
+                        ${{ runner.os }}-maven-
+
+            -   name: Cache android jar
+                id: android-cache
+                uses: actions/cache@v1
+                with:
+                    path: android-jar-cache
+                    key: ${{ runner.os }}-android-version-${{matrix.java_version}}-${{ env.android-version }}
```

Comment:
what is the reason that you want to cache the android.jar per java version? It's always the same and thus could be shared no?

Reply:
I thought of future support where we might want to differentiate processes according to java versions
It can be removed if not relevant

Reply:
I would remove it and add later once required